cfbs/analyze.py

```diff
@@ -0,0 +1,597 @@
+    norm_path = path.replace(os.sep, "/")
+
+    if is_parentpath:
+        if norm_path.startswith(masterfiles_dir + "/"):
+            norm_path = os.path.relpath(norm_path, masterfiles_dir)
+            # `os.path.relpath` will still output paths with `os.sep`, even if `norm_path` uses forward slashes on e.g. Windows
+            norm_path = norm_path.replace(os.sep, "/")
+            norm_path = "masterfiles/" + norm_path
+    else:
+        norm_path = "masterfiles/" + norm_path
+
+    return norm_path
+
+
+def mpf_denormalized_path(path, is_parentpath, masterfiles_dir):
+    """Inverse function of `mpf_normalized_path`."""
+    denorm_path = path
+    # this does work as intended even if the first dir isn't masterfiles and there's a masterfiles dir deeper in the path
+    relpath = os.path.relpath(denorm_path, "masterfiles")
+
+    if is_parentpath:
+        # if `"masterfiles"`, substitute to `masterfiles_dir`
+        # if not, then the path should stay the same
+        if not relpath.startswith(".." + os.sep):
+            denorm_path = os.path.join(masterfiles_dir, relpath)
+
+    else:
+        # this will work as intended even for other directories than `masterfiles` e.g. `modules`
+        denorm_path = relpath
+
+    return denorm_path
+
+
+class VersionsCounter:
+    def __init__(self):
+        self._versions_counts = {}
+
+    def increment(self, version):
+        if version not in self._versions_counts:
+            self._versions_counts[version] = 0
+        self._versions_counts[version] += 1
+
+    def most_common_version(self):
+        """Returns version with the highest count. In case of a tie, returns the highest version with the highest count."""
+        highest_count = max(self._versions_counts.values(), default=0)
+
+        versions_with_highest_count = [
+            k for (k, v) in self._versions_counts.items() if v == highest_count
+        ]
+
+        return highest_version(versions_with_highest_count)
+
+    def sorted_list(self):
+        """Returns a sorted list of key-value pairs `(version, count)`. The sorting is in descending order. In case of a count tie, the higher version's pair is considered greater."""
+        return sorted(
+            self._versions_counts.items(),
+            key=lambda item: (item[1], version_as_comparable_list(item[0])),
+            reverse=True,
+        )
+
+    def is_empty(self):
+        return self._versions_counts == {}
+
+
+class VersionsData:
+    def __init__(self):
+        self.version_counter = VersionsCounter()
+        self.highest_version_counter = VersionsCounter()
+        # acronyms: vc = version_counter, hvc = highest_version_counter
+        self.different_filepath_vc = VersionsCounter()
+        self.different_filepath_hvc = VersionsCounter()
+
+    def display(self):
+        if not self.version_counter.is_empty():
+            print(
+                "Same filepath versions distribution:",
```

Comment:
From looking at just the output from the command, it is difficult to understand what this is

Reply:
Is there an example of the output here or in the man page contents?